sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5364,19 +5434,15 @@ pi_result piEventGetProfilingInfo(pi_event Event, pi_profiling_info ParamName,
     return PI_PROFILING_INFO_NOT_AVAILABLE;
   }
 
-  uint64_t ZeTimerResolution =
-      Event->Queue
-          ? Event->Queue->Device->ZeDeviceProperties->timerResolution
-          : Event->Context->Devices[0]->ZeDeviceProperties->timerResolution;
+  auto Device =
+      Event->Queue ? Event->Queue->Device : Event->Context->Devices[0];
+
+  std::shared_lock Lock(Device->Mutex);
```

Comment:
Is this to protect access to Device->ZeDeviceProperties?
This field is of ZeCache type, which is written only once, and then accessed read-only. Is this lock really sufficient?
Maybe we should hold a unique lock when writing it, and then no lock?

Reply:
If it is written only once, then why should we have unique_lock when writing then?

Reply:
> If it is written only once, then could you please clarify why should we have unique_lock when writing?

To guarantee that it is initialized only once, i.e. no other thread attempts to write it while first one is doing so already.

Reply:
Access to ZeCache using -> can trigger computation of the property and we don't know when this is going to happen, so we need to use unique_lock for each access because each access can potentially do write.